### PR TITLE
Add key rotation and fix client version

### PR DESCRIPTION
**Pull request by jsupun**
Added a handler for http error/exceptions. It is for both
the get_secrets() and save() method. The exception handler will
format errors messages and throw the correct exception. It will
also handle the key rotation if it gets an 'key' error.

If a key error happens, the public key in the dictionary with
the server provided key_id will be used. The config is updated
with this key_id. The request that caused the excpetion will be
retried.

### CODE REVIEW

File: sdk/python/core/keeper_secrets_manager_core/core.py [hunk 1]
```diff
@@ -273,53 +284,75 @@ def _post_query(self, path, context, payload_and_signature):
...
         return rs
 
+    def handler_http_error(self, rs):
+
+        try:
+            # Decode the JSON content, throw exception if not JSON
+            response_dict = json_to_dict(rs.text)
+            if response_dict is None:
+                raise json.JSONDecodeError("Not JSOM", "NONE", 0)
+
```
Comment (maksimu) [inline sdk/python/core/keeper_secrets_manager_core/core.py:294]: JSOM?

Comment (jsupun) [reply sdk/python/core/keeper_secrets_manager_core/core.py:294]: It's the budget off-brand of JSON

File: sdk/python/core/keeper_secrets_manager_core/core.py [hunk 2]
```diff
@@ -273,54 +284,75 @@ def _post_query(self, path, context, payload_and_signature):
...
+            if error == 'invalid_client_version':
+                logging.error("Client version {} was not registered in the backend".format(
+                    keeper_secrets_manager_sdk_client_id))
+                msg = response_dict.get('additional_info')
+
+            # The server wants us to use a different public key.
+            elif error == 'key':
+                key_id = response_dict.get("key_id")
+                logging.info("Server has requested we public key {}".format(key_id))
```
Comment (maksimu) [inline sdk/python/core/keeper_secrets_manager_core/core.py]: you meant to say "Server has requested we **use** public key" ?

File: sdk/python/core/keeper_secrets_manager_core/core.py [hunk 3]
```diff
@@ -273,54 +284,75 @@ def _post_query(self, path, context, payload_and_signature):
...
+
+                if key_id is None:
+                    raise ValueError("The public key is is blank from the server")
```
Comment (maksimu) [inline sdk/python/core/keeper_secrets_manager_core/core.py]: typo

File: sdk/python/core/keeper_secrets_manager_core/core.py (hunk 1 above)
Comment (maksimu) [reply sdk/python/core/keeper_secrets_manager_core/core.py:294]: 💸

**Pull request merged**